src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/sdpa_fuse_transpose_reshape.cpp

```diff
@@ -0,0 +1,200 @@
+ *                   |                                       output
+ *               out_reshpae
+ *                   |
+ *                 output
+ */
+
+using namespace ov;
+using namespace ov::pass::pattern;
+
+intel_cpu::SDPAFuseTransposeReshape::SDPAFuseTransposeReshape() {
+    MATCHER_SCOPE(SDPAFuseTransposeReshape);
+
+    auto q_reshape_node = wrap_type<op::v1::Reshape>({any_input(), any_input()});
+    auto k_reshape_node = wrap_type<op::v1::Reshape>({any_input(), any_input()});
+    auto v_reshape_node = wrap_type<op::v1::Reshape>({any_input(), any_input()});
+
+    auto q_transpose_order_node = wrap_type<op::v0::Constant>();
+    auto k_transpose_order_node = wrap_type<op::v0::Constant>();
+    auto v_transpose_order_node = wrap_type<op::v0::Constant>();
+    auto q_transpose_node = wrap_type<op::v1::Transpose>({q_reshape_node, q_transpose_order_node});
+    auto k_transpose_node = wrap_type<op::v1::Transpose>({k_reshape_node, k_transpose_order_node});
+    auto v_transpose_node = wrap_type<op::v1::Transpose>({v_reshape_node, v_transpose_order_node});
+
+    auto sdpa_node =
+        wrap_type<op::v13::ScaledDotProductAttention>({q_transpose_node, k_transpose_node, v_transpose_node});
+
+    auto out_transpose_order_node = wrap_type<op::v0::Constant>();
+    auto out_transpose_node = wrap_type<op::v1::Transpose>({sdpa_node, out_transpose_order_node});
+    auto out_reshape_node = wrap_type<op::v1::Reshape>({out_transpose_node, wrap_type<op::v0::Constant>()});
+
+    matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](pass::pattern::Matcher& m) {
+        auto& pattern_map = m.get_pattern_value_map();
+        auto sdpa = as_type_ptr<op::v13::ScaledDotProductAttention>(pattern_map.at(sdpa_node).get_node_shared_ptr());
+        if (sdpa == nullptr || transformation_callback(sdpa)) {
+            return false;
+        }
+
+        // Order=[0, 2, 1, 3]
+        auto is_expected_transpose = [&](std::shared_ptr<op::v1::Transpose>& transpose) {
+            if (transpose) {
+                const auto orders = as_type_ptr<op::v0::Constant>(transpose->get_input_node_shared_ptr(1));
+                return orders && (std::vector<int32_t>({0, 2, 1, 3}) == orders->cast_vector<int32_t>());
+            }
+            return false;
+        };
+
+        // Reshape [B,L,H*S] -> [B,L,H,S]
+        auto is_expected_reshape = [&](std::shared_ptr<op::v1::Reshape>& reshape_node, bool reverse = false) {
+            if (reshape_node) {
+                if (reshape_node->get_input_size() != 2u) {
+                    return false;
+                }
+                auto inp_shape = reshape_node->get_input_partial_shape(0);
+                auto outp_shape = reshape_node->get_output_partial_shape(0);
+                // Expect shape: [?, ?, val]
+                auto check_dim_3 = [](ov::PartialShape shape) {
+                    return shape.rank() == 3 && shape[2].is_static();
```

Comment:
Better to add `shape.rank().is_static()` check here.

Reply:
Done